`input.validity` reports `valid: true` for partially completed dates/times

nt1m opened the pull request:
#### ec9036933491f8f64f0f2c1c75736a2e125da1b9
<pre>
`input.validity` reports `valid: true` for partially completed dates/times
<a href="https://bugs.webkit.org/show_bug.cgi?id=248401">https://bugs.webkit.org/show_bug.cgi?id=248401</a>
rdar://102984901

Reviewed by Aditya Keerthi.

Porting over parts of: <a href="https://github.com/chromium/chromium/commit/f890442a85ed9791096348138eb305cd97b385b7">https://github.com/chromium/chromium/commit/f890442a85ed9791096348138eb305cd97b385b7</a>

We should report badInput: true (and hence valid: false), when the input[type=date/time/datetime-local] are partially filled.

* LayoutTests/fast/forms/date/date-validity-badinput-expected.txt: Added.
* LayoutTests/fast/forms/date/date-validity-badinput.html: Added.
* LayoutTests/fast/forms/datetimelocal/datetimelocal-validity-badinput-expected.txt: Added.
* LayoutTests/fast/forms/datetimelocal/datetimelocal-validity-badinput.html: Added.
* LayoutTests/fast/forms/resources/datetime-validity-badinput.js: Added.
(testBadInput.test):
(testBadInput):
* LayoutTests/fast/forms/time/time-validity-badinput-expected.txt: Added.
* LayoutTests/fast/forms/time/time-validity-badinput.html: Added.
* LayoutTests/platform/ios-wk2/TestExpectations:
* Source/WebCore/html/BaseDateAndTimeInputType.cpp:
(WebCore::BaseDateAndTimeInputType::hasBadInput const):
* Source/WebCore/html/BaseDateAndTimeInputType.h:
* Source/WebCore/html/shadow/DateTimeEditElement.cpp:
(WebCore::DateTimeEditElement::editableFieldsHaveValues const):
* Source/WebCore/html/shadow/DateTimeEditElement.h:

Canonical link: <a href="https://commits.webkit.org/263748@main">https://commits.webkit.org/263748@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/da356c9b11f09f0eef4b05bd383aeefc1df8fba1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/5674 "Failed to checkout and rebase branch from PR 13523") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/5830 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/6020 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/7226 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/5677 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/9/builds/5675 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/6052 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/5802 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/7826 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [❌ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/5780 "Failed to checkout and rebase branch from PR 13523") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/5826 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/5083 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/7275 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/3303 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/5107 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/12805 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/5171 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/5184 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/7024 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/5623 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/4600 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/30/builds/5070 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/5037 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/9184 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/649 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/5431 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

pxlcoder commented on Source/WebCore/html/shadow/DateTimeEditElement.cpp:
```diff
@@ -404,6 +404,15 @@ DateTimeFieldsState DateTimeEditElement::valueAsDateTimeFieldsState() const
+bool DateTimeEditElement::editableFieldsHaveValues() const
+{
+    for (auto& field : m_fields) {
+        if (!field->isDisabledFormControl() && field->hasValue())
```
The fields are just `HTMLDivElement`, so `isDisabledFormControl()` is always going to be false?

nt1m replied:
Removed the `isDisabledFormControl()`, this was copy pasta from Chromium

nt1m committed:
`input.validity` reports `valid: true` for partially completed dates/times
https://bugs.webkit.org/show_bug.cgi?id=248401
rdar://102984901

Reviewed by Aditya Keerthi.

Porting over parts of: https://github.com/chromium/chromium/commit/f890442a85ed9791096348138eb305cd97b385b7

We should report badInput: true (and hence valid: false), when the input[type=date/time/datetime-local] are partially filled.

* LayoutTests/fast/forms/date/date-validity-badinput-expected.txt: Added.
* LayoutTests/fast/forms/date/date-validity-badinput.html: Added.
* LayoutTests/fast/forms/datetimelocal/datetimelocal-validity-badinput-expected.txt: Added.
* LayoutTests/fast/forms/datetimelocal/datetimelocal-validity-badinput.html: Added.
* LayoutTests/fast/forms/resources/datetime-validity-badinput.js: Added.
(testBadInput.test):
(testBadInput):
* LayoutTests/fast/forms/time/time-validity-badinput-expected.txt: Added.
* LayoutTests/fast/forms/time/time-validity-badinput.html: Added.
* LayoutTests/platform/ios-wk2/TestExpectations:
* Source/WebCore/html/BaseDateAndTimeInputType.cpp:
(WebCore::BaseDateAndTimeInputType::hasBadInput const):
* Source/WebCore/html/BaseDateAndTimeInputType.h:
* Source/WebCore/html/shadow/DateTimeEditElement.cpp:
(WebCore::DateTimeEditElement::editableFieldsHaveValues const):
* Source/WebCore/html/shadow/DateTimeEditElement.h:

Canonical link: https://commits.webkit.org/263748@main
